src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -327,7 +328,28 @@ async def generate_lockfiles_goal(
     union_membership: UnionMembership,
     generate_lockfiles_subsystem: GenerateLockfilesSubsystem,
     python_setup: PythonSetup,
+    python_repos: PythonRepos,
 ) -> GenerateLockfilesGoal:
+    failure_instructions = (
+        "If lockfile generation fails, you can either disable lockfiles by setting "
+        "`[tool].lockfile = '<none>'`, e.g. setting `[black].lockfile`, or you can manually "
+        "generate lockfiles, e.g. by using pip-compile or `pip freeze`. If manually generating "
+        "lockfiles, you should set `[python-setup].invalid_lockfile_behavior = 'ignore'."
+    )
+    if python_repos.repos:
+        logger.warning(
+            "The option `[python-repos].repos` is configured, but it does not currently work with "
+            "lockfile generation because Pants is currently using Poetry for lockfile generation "
+            "and Poetry does not support `--find-links`. Lockfile generation may fail.\n\n"
```

Comment:
Although the Poetry shoutout is admirable, explaining "why" find-links isn't supported probably isn't necessary here. Would prune that bit to make this more concise.